CMakeLists.txt

```diff
@@ -37,8 +37,21 @@ endif()
 # Boost package
 ###############################################################################
 
+set(Boost_COMPONENTS coroutine context system thread)
+
+# If on Windows, rather than be annoyed by constant failure to find DLL errors
+# during test debugging simply force all static linking of Boost libraries
+if(WIN32 AND NOT MINGW)
+  set(Boost_USE_STATIC_LIBS ON)
+else()
+  set(Boost_COMPONENTS ${Boost_COMPONENTS} unit_test_framework)
```

Comment:
Oh Linux can be header only too of course. It's just you don't need it as much there because ELF SOs encode the path to their dependencies, so when you run the test executable by hand/from gdb it all "just works" usually.

Windows, on the other hand, produces executables which won't execute by default under your configuration above unless you fiddle PATH first. This was costing me productivity. So I changed the default.

If I'm going ahead with MAID-855 then all this is moot anyway as a Jamfile replaces cmake here entirely. Then with ./b2 you simply choose link=static on Windows, and it's done.